pkg/kubelet/secret/secret_manager.go

```diff
@@ -201,8 +201,12 @@ type cachingSecretManager struct {
 }
 
 func NewCachingSecretManager(kubeClient clientset.Interface) (Manager, error) {
+	// TODO: Currently apiserver completely cannot keep up with 1 minute ttl in
+	// 5000-node clusters. So before we optimize apiserver and/or etcd, increase
+	// the TTL to 5 minutes (which is a lot, but...)
+	ttl := 5 * time.Minute
```

Comment:
> I think that having the "SecretManager" abstraction is useful anyway, even if we decide to change the implementation completely in few weeks.

Agreed. I am just curious about the cause of the "improvement".

One more question. Is the problem that nodes send too many requests at the same time (which can be mitigated by adding jitters), or is the load just too high even though the requests are spread out?

> But going back - does increasing the ttl to e.g. 2m30s or sth like that sounds reasonable to you?

This is a user-facing change, and probably demands more discussion. Maybe we shouldn't assume that one configuration can work with all cluster sizes. Can we raise the TTL only in the 5000-node cluster and document it (until kubelet can watch secrets)?